### PR TITLE
Add coo2csr

### PR DESCRIPTION
**Pull request by e10harvey**
Related to #1164.

Note to reviewers: There is a work around in place for a suspected atomics bug in scratch space. The work around ensures that all hashmaps / usets in scratch memory have room for duplicate keys rather than sizing these to the sparsity calculations. Once the atomics + scratch problem is addressed, I will update the scratch memory allocations to use the sparsity calculations.

### CODE REVIEW

File: common/src/KokkosKernels_HashmapAccumulator.hpp [hunk 1]
```diff
@@ -730,6 +731,330 @@ struct HashmapAccumulator {
...
+        return __insert_success;
+      }
+    }
+
+    // Insert either a new or collided key. Relaxed memory order is OK. We have
+    // RAW dependencies below.
+    my_write_index = Kokkos::atomic_fetch_add(used_size_, size_type(1));
+
+// clang-format off
+#if 0
```
Comment (srajama1) [inline common/src/KokkosKernels_HashmapAccumulator.hpp:777]: Please remove dead code.

File: common/src/KokkosKernels_HashmapAccumulator.hpp [hunk 2]
```diff
@@ -730,6 +731,330 @@ struct HashmapAccumulator {
...
+      }
+    }
+#endif
+    // clang-format on
+
+    if (my_write_index >= __max_value_size) {
+#ifdef HASHMAPACCUMULATOR_ASSERT_ENABLED
+      for (i = 0; i < __max_value_size; i++) {
+        if (Kokkos::volatile_load(&keys[i]) == key) {
+          printf("head: %d, i: %d, key: %lld\n", head, i, key);
```
Comment (srajama1) [inline common/src/KokkosKernels_HashmapAccumulator.hpp:790]: Does printf work inside KOKKOS_INLINE_FUNCTION ? How does this even work?

File: common/src/KokkosKernels_HashmapAccumulator.hpp [hunk 3]
```diff
@@ -730,6 +731,330 @@ struct HashmapAccumulator {
...
+        return i;
+      }
+    }
+
+    // Insert either a new or collided key. Relaxed memory order is OK. We have
+    // RAW dependencies below.
+    my_write_index = Kokkos::atomic_fetch_add(used_size_, size_type(1));
+
+// clang-format off
+#if 0
```
Comment (srajama1) [inline common/src/KokkosKernels_HashmapAccumulator.hpp:854]: May be this should be a verbose option instead of if 0

**Review by srajama1 — APPROVED** [review]
Couple of small comments.

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pull Request AutoTester' - Failure: Timed out waiting for job KokkosKernels_PullRequest_GCC930_Light_Tpls_GCC930 to start: Total Wait = 3603
- Other jobs have been previously started - We must stop them...

**Comment by e10harvey** [comment]
Closing. There is a data race in the hashmapAccumulator that could not be addressed due to the unstructured nature of the coo input. In short, the same key can be inserted by different threads.

**Pull request closed without merging**